### PR TITLE
CLI Watcher

### PR DESCRIPTION
**Pull request by expede**
Pretty much what it says on the tin. CLI now has a watcher. Also fixed some bug on master that... broke `fission up`? I dunno, it was fixed on this branch before we found it in production.

```bash
# You can now do this:
fission-cli watch
```

### CODE REVIEW

**Comment by expede** [comment]
Pics or it didn't happen:

Screen Shot 2019-09-30 at 14 20 17

File: library/Fission/CLI/Success.hs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+live :: MonadIO m => Text -> m ()
+live hash = do
+  UTF8.putText $ Emoji.rocket <> "Now live on the network\n"
+  UTF8.putText $ Emoji.okHand <> hash  <> "\n"
+
+putOk :: MonadIO m => Text -> m ()
+putOk msg = do
+  liftIO $ ANSI.setSGR [ANSI.SetColor ANSI.Foreground ANSI.Vivid ANSI.Green]
+  UTF8.putText $ Emoji.whiteHeavyCheckMark <> " " <> msg
```
Comment (dholms) [inline library/Fission/CLI/Success.hs]: please reset ANSI. terminal forever green

File: library/Fission/CLI/Watch.hs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        => Has IPFS.Timeout  cfg
+        => m ()
+watcher = do
+  cfg <- ask
+  dir <- getCurrentDirectory
+  UTF8.putText $ Emoji.eyes <> " Watching " <> Text.pack dir <> " for changes...\n"
+
+  IPFS.addDir dir >>= \case
+    Left err ->
+      logError $ displayShow err
```
Comment (dholms) [inline library/Fission/CLI/Watch.hs]: Use the CLI.Error instead of logger

File: library/Fission/CLI/Pin.hs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  logDebug $ "Remote pinning " <> display hash
+  Client.Runner runner <- Config.get
+
+  pin runner auth cid >>= \case
+    Right _ ->
+      success
+
+    Left _ -> do
+      logError "Failed to pin remotely, attempting to reconnect to IPFS"
+      IPFS.Peer.connect IPFS.Peer.fission
```
Comment (dholms) [inline library/Fission/CLI/Pin.hs]: do a connect before trying initial pin as well

**Pull request merged**